armory/scenarios/poison.py

```diff
@@ -283,33 +321,114 @@ def load_dataset(self, eval_split_default="test"):
             **self.dataset_kwargs,
         )
         self.i = -1
+        self.test_set_class_labels = set()
 
     def load_metrics(self):
-        self.accuracy_on_benign_data_all_classes = metrics.MetricList(
-            "categorical_accuracy"
+        if self.use_filtering_defense:
+            # Filtering metrics
+            self.hub.connect_meter(
+                Meter(
+                    "filter",
+                    metrics.get_supported_metric("tpr_fpr"),
+                    "scenario.poisoned",
+                    "scenario.removed",
+                )
+            )
+
+        self.hub.connect_meter(
+            Meter(
+                "sample_accuracy_on_benign_test_data_all_classes",
+                get_supported_metric("categorical_accuracy"),
+                "scenario.y[benign]",
```

Comment:
This one didn't work: `Meter 'sample_accuracy_on_benign_test_data_all_classes' was never measured. The following args were never set: ['scenario.y[benign]']`  

(`sample_accuracy_on_poisoned_data_all_classes` had the same error but I think will have the same fix)

Reply:
I removed the stage filters, as they are unnecesary, which should fix the problem.